#VRML_SIM R2019a utf8
WorldInfo {
}
Viewpoint {
  orientation 0.15265288632430604 -0.9845185869385995 -0.08614086294716126 5.242275813843718
  position 3.7967066820991024 1.352179364420509 1.9862251897166063
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
Robot {
  translation 0 0.10000000000000002 0
  rotation 0 1 0 1.0471975479999998
  children [
    DEF BODY Shape {
      appearance DEF M_GREEN PBRAppearance {
        baseColor 0.010070954451819639 0.5033340962844282 0.15786984054322117
        roughness 1
      }
      geometry Cylinder {
        height 0.08
        radius 0.14
      }
    }
    DEF HEAD_BASE1 Transform {
      translation 0 0.05 0
      children [
        Shape {
          appearance USE M_GREEN
          geometry Cylinder {
            height 0.1
            radius 0.12
          }
        }
      ]
    }
    DEF HEAD_BASE2 Transform {
      translation 0 0.12999999999999998 0
      children [
        Shape {
          appearance USE M_GREEN
          geometry Cylinder {
            height 0.1
            radius 0.1
          }
        }
        HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 1 0
          }
          device [
            PositionSensor {
              name ""
            }
            RotationalMotor {
              name "motor_ds"
            }
          ]
          endPoint Solid {
            translation 0.07000000019179575 0.020000002589244142 0
            rotation 0 0 1 3.141592644
            children [
              Transform {
                children [
                  Shape {
                    appearance PBRAppearance {
                      baseColor 0 0 0
                    }
                    geometry Box {
                      size 0.1 0.04 0.1
                    }
                  }
                  DistanceSensor {
                    translation -0.04999999108149206 -3.4523258188459517e-09 0
                    rotation 0 0 -1 3.1415926823591733
                    children [
                      Shape {
                        appearance PBRAppearance {
                          baseColor 0.9333333333333333 0.33600366216525523 0.33600366216525523
                          roughness 1
                          metalness 0
                        }
                        geometry Box {
                          size 0.02 0.02 0.02
                        }
                      }
                    ]
                  }
                ]
              }
            ]
            name "solid(4)"
            physics Physics {
            }
          }
        }
      ]
    }
    DEF HEAD_BALL Transform {
      translation 0 0.18999999999999995 0
      children [
        Shape {
          appearance USE M_GREEN
          geometry Sphere {
            radius 0.09
          }
        }
        HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 1 0
            anchor 0 0.08 0
          }
          device [
            RotationalMotor {
              name "motor_gun"
            }
            PositionSensor {
              name "ps_motor"
            }
          ]
          endPoint Solid {
            translation 0.16000000000000003 0.04000000000000003 0
            children [
              Shape {
                appearance USE M_GREEN
                geometry DEF SHOOTER Box {
                  size 0.2 0.03 0.04
                }
              }
            ]
            name "solid(3)"
            physics Physics {
            }
          }
        }
      ]
    }
    DEF WH_1 Transform {
      translation 0 0 0.14
      children [
        DEF WHEEL1 HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 0 1
            anchor 0 0 0.02
          }
          device [
            RotationalMotor {
              name "wheel1"
              maxVelocity 30.3687
              maxTorque 0.8473
            }
            PositionSensor {
              name "ps_1"
              resolution 0.003848
            }
          ]
          endPoint Solid {
            translation 0 0 -0.009999999999999998
            rotation 0 0 1 0
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 0.06 0.02
                }
                endPoint Solid {
                  translation 0 0.06 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r1 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        DEF SMALL_WHEEL_SHAPE Shape {
                          appearance PBRAppearance {
                            baseColor 1 0.752941 0.796078
                            roughness 1
                            metalness 0
                          }
                          geometry Cylinder {
                            height 0.03
                            radius 0.012
                          }
                        }
                      ]
                    }
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor -0.04535 0.04535 0.02
                }
                endPoint Solid {
                  translation -0.04535 0.04535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r2 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 -0.06 0.02
                }
                endPoint Solid {
                  translation 0 -0.06 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r3 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor 0.04535 -0.04535 0.02
                }
                endPoint Solid {
                  translation 0.04535 -0.04535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r4 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor -0.06 0 0.02
                }
                endPoint Solid {
                  translation -0.06 0 0.02
                  children [
                    DEF r5 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor 0.06 0 0.02
                }
                endPoint Solid {
                  translation 0.06 0 0.02
                  children [
                    DEF r6 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor -0.04535 -0.04535 0.02
                }
                endPoint Solid {
                  translation -0.04535 -0.04535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r7 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor 0.04535 0.04535 0.02
                }
                endPoint Solid {
                  translation 0.04535 0.04535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r8 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF WHEEL_TRANS Transform {
                translation 0 0 0.02
                rotation 1 0 0 1.5708
                children [
                  DEF WHEEL_SHAPE Shape {
                    appearance PBRAppearance {
                      baseColor 0.9333333333333333 0.33600366216525523 0.33600366216525523
                      roughness 1
                    }
                    geometry Cylinder {
                      height 0.02
                      radius 0.06
                      subdivision 20
                    }
                  }
                ]
              }
            ]
            boundingObject USE WHEEL_TRANS
            physics Physics {
            }
          }
        }
      ]
    }
    DEF WH_2 Transform {
      translation 0.13 0 -0.069
      rotation 0 1 0 2.094395096
      children [
        DEF WHEEL1 HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 0 1
            anchor 0 0 0.02
          }
          device [
            RotationalMotor {
              name "wheel2"
              maxVelocity 30.3687
              maxTorque 0.8473
            }
            PositionSensor {
              name "ps_2"
              resolution 0.003848
            }
          ]
          endPoint Solid {
            translation 0 0 -0.009999999999999998
            rotation 0 0 1 0
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 0.06 0.02
                }
                endPoint Solid {
                  translation 0 0.06 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r1 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        DEF SMALL_WHEEL_SHAPE Shape {
                          appearance PBRAppearance {
                            baseColor 1 0.752941 0.796078
                            roughness 1
                            metalness 0
                          }
                          geometry Cylinder {
                            height 0.03
                            radius 0.012
                          }
                        }
                      ]
                    }
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor -0.04535 0.04535 0.02
                }
                endPoint Solid {
                  translation -0.04535 0.04535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r2 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 -0.06 0.02
                }
                endPoint Solid {
                  translation 0 -0.06 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r3 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor 0.04535 -0.04535 0.02
                }
                endPoint Solid {
                  translation 0.04535 -0.04535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r4 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor -0.06 0 0.02
                }
                endPoint Solid {
                  translation -0.06 0 0.02
                  children [
                    DEF r5 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor 0.06 0 0.02
                }
                endPoint Solid {
                  translation 0.06 0 0.02
                  children [
                    DEF r6 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor -0.04535 -0.04535 0.02
                }
                endPoint Solid {
                  translation -0.04535 -0.04535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r7 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor 0.04535 0.04535 0.02
                }
                endPoint Solid {
                  translation 0.04535 0.04535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r8 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF WHEEL_TRANS Transform {
                translation 0 0 0.02
                rotation 1 0 0 1.5708
                children [
                  USE WHEEL_SHAPE
                ]
              }
            ]
            name "solid(1)"
            boundingObject USE WHEEL_TRANS
            physics Physics {
            }
          }
        }
      ]
    }
    DEF WH_3 Transform {
      translation -0.13 0 -0.069
      rotation 0 0.9999999999999999 0 -2.0944
      children [
        DEF WHEEL1 HingeJoint {
          jointParameters HingeJointParameters {
            axis 0 0 1
            anchor 0 0 0.02
          }
          device [
            RotationalMotor {
              name "wheel3"
              maxVelocity 30.3687
              maxTorque 0.8473
            }
            PositionSensor {
              name "ps_3"
              resolution 0.003848
            }
          ]
          endPoint Solid {
            translation 0 0 -0.009999999999999998
            rotation 0 0 1 0
            children [
              DEF sr1 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 0.06 0.02
                }
                endPoint Solid {
                  translation 0 0.06 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r1 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        DEF SMALL_WHEEL_SHAPE Shape {
                          appearance PBRAppearance {
                            baseColor 1 0.752941 0.796078
                            roughness 1
                            metalness 0
                          }
                          geometry Cylinder {
                            height 0.03
                            radius 0.012
                          }
                        }
                      ]
                    }
                  ]
                  name "sr1"
                  boundingObject USE r1
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr2 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor -0.04535 0.04535 0.02
                }
                endPoint Solid {
                  translation -0.04535 0.04535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r2 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr2"
                  boundingObject USE r2
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr3 HingeJoint {
                jointParameters HingeJointParameters {
                  anchor 0 -0.06 0.02
                }
                endPoint Solid {
                  translation 0 -0.06 0.02
                  rotation 1 0 0 0
                  children [
                    DEF r3 Transform {
                      rotation 0 0 1 1.5708
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr3"
                  boundingObject USE r3
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr4 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0.707107 0.707107 0
                  anchor 0.04535 -0.04535 0.02
                }
                endPoint Solid {
                  translation 0.04535 -0.04535 0.02
                  rotation 0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r4 Transform {
                      rotation 0 0 1 2.27
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr4"
                  boundingObject USE r4
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr5 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor -0.06 0 0.02
                }
                endPoint Solid {
                  translation -0.06 0 0.02
                  children [
                    DEF r5 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr5"
                  boundingObject USE r5
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr6 HingeJoint {
                jointParameters HingeJointParameters {
                  axis 0 1 0
                  anchor 0.06 0 0.02
                }
                endPoint Solid {
                  translation 0.06 0 0.02
                  children [
                    DEF r6 Transform {
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr6"
                  boundingObject USE r6
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr7 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor -0.04535 -0.04535 0.02
                }
                endPoint Solid {
                  translation -0.04535 -0.04535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r7 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr7"
                  boundingObject USE r7
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF sr8 HingeJoint {
                jointParameters HingeJointParameters {
                  axis -0.707107 0.707107 0
                  anchor 0.04535 0.04535 0.02
                }
                endPoint Solid {
                  translation 0.04535 0.04535 0.02
                  rotation -0.7071067811865476 0.7071067811865476 0 0
                  children [
                    DEF r8 Transform {
                      rotation 0 0 1 0.76
                      children [
                        USE SMALL_WHEEL_SHAPE
                      ]
                    }
                  ]
                  name "sr8"
                  boundingObject USE r8
                  physics Physics {
                    density -1
                    mass 0.1
                  }
                }
              }
              DEF WHEEL_TRANS Transform {
                translation 0 0 0.02
                rotation 1 0 0 1.5708
                children [
                  USE WHEEL_SHAPE
                ]
              }
            ]
            name "solid(2)"
            boundingObject USE WHEEL_TRANS
            physics Physics {
            }
          }
        }
      ]
    }
  ]
  boundingObject USE BODY
  physics Physics {
  }
  controller "u3_real_exam_munoz_samara"
}
